File: Lab3OOP/MusicCollection.cpp
#include "MusicCollection.h"
#include "Song.h"
#include "InstrumentalWork.h"
#include <iostream>
#include <fstream>
#include "UserInterface.h"

MusicCollection::MusicCollection() {}

const std::vector<Composition*>& MusicCollection::getCompositions() const {
    return compositions;
}

MusicCollection::~MusicCollection() {
    for (auto composition : compositions) {
        delete composition;
    }
}


void MusicCollection::addComposition(Composition* composition) {
    compositions.push_back(composition);
}

bool MusicCollection::removeComposition(size_t index) {
    if (index < compositions.size()) {
        compositions.erase(compositions.begin() + index);
        return true; 
    }
    return false; 
}

File: Lab3_oop_python/music_collection.py
from song import Song
from instrumental_work import InstrumentalWork
from composition import Composition

class MusicCollection:
    def __init__(self):
        self.compositions = []

    def add_composition(self, composition: Composition):
        self.compositions.append(composition)

    def remove_composition(self, index: int) -> bool:
        if 0 <= index < len(self.compositions):
            del self.compositions[index]
            return True
        return False

    def get_compositions(self) -> list:
        return self.compositions
    
